homeassistant/components/bluetooth/scanner.py

```diff
@@ -199,48 +251,78 @@ async def _async_start(self) -> None:
     @hass_callback
     def _async_setup_scanner_watchdog(self) -> None:
         """If Dbus gets restarted or updated, we need to restart the scanner."""
-        self._last_detection = MONOTONIC_TIME()
-        self._cancel_watchdog = async_track_time_interval(
-            self.hass, self._async_scanner_watchdog, SCANNER_WATCHDOG_INTERVAL
-        )
+        self._start_time = self._last_detection = MONOTONIC_TIME()
+        if not self._cancel_watchdog:
+            self._cancel_watchdog = async_track_time_interval(
+                self.hass, self._async_scanner_watchdog, SCANNER_WATCHDOG_INTERVAL
+            )
 
     async def _async_scanner_watchdog(self, now: datetime) -> None:
         """Check if the scanner is running."""
         time_since_last_detection = MONOTONIC_TIME() - self._last_detection
+        _LOGGER.debug(
+            "%s: Scanner watchdog time_since_last_detection: %s",
+            self.name,
+            time_since_last_detection,
+        )
         if time_since_last_detection < SCANNER_WATCHDOG_TIMEOUT:
             return
         _LOGGER.info(
-            "%s: Bluetooth scanner has gone quiet for %s, restarting",
+            "%s: Bluetooth scanner has gone quiet for %ss, restarting",
             self.name,
-            SCANNER_WATCHDOG_INTERVAL,
+            SCANNER_WATCHDOG_TIMEOUT,
         )
         async with self._start_stop_lock:
-            await self._async_stop()
-            await self._async_start()
+            # Stop the scanner but not the watchdog
+            # since we want to try again later if it's still quiet
+            await self._async_stop_scanner()
+            if self._start_time == self._last_detection or (
+                time_since_last_detection
+            ) > (SCANNER_WATCHDOG_TIMEOUT + SCANNER_WATCHDOG_INTERVAL.total_seconds()):
```

Comment:
Are both of these const? Probably not a big deal, but should we calculate this once?

Reply:
Hopefully its not called that often :)

Reply:
It would be nice to have it as a named constant though

Reply:
https://github.com/home-assistant/core/pull/77043#discussion_r950746953